src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -174,15 +199,21 @@ public void RemoveAt(int index)
         {
             get
             {
-                if (index >= _size)
+                // Following trick can reduce the range check by one comparison
+                if ((uint)index >= (uint)Count)
```

Comment:
We're talking past each other.  I understand what your change does.  I understand that your change is fixing something mentioned in that issue.  I'm saying that issue was representing just one person's opinion of a change that could be made, not an agreement that it's a change that should actually be made.  The change is breaking because it ends up throwing one exception type in a situation where it used to throw a different exception type, and such changes are not taken lightly.  We do take such breaking changes when there is significant value in doing so, when that value outweighs the potential for a break.  I currently see close to no value in making such a change, which to me means it's not a good idea to change.